tests/TestCase/Database/QueryCompilerTest.php

```diff
@@ -0,0 +1,225 @@
+    }
+
+    public function testUpdate(): void
+    {
+        /** @var \Cake\Database\Query\UpdateQuery $query */
+        $query = $this->newQuery(Query::TYPE_UPDATE);
+        $query = $query->update('articles')
+            ->set('name', 'mark')
+            ->where(['id' => 1]);
+        $result = $this->compiler->compile($query, $this->binder);
+        $this->assertSame('UPDATE articles SET name = :c0 WHERE id = :c1', $result);
+    }
+
+    public function testUpdateWithComment(): void
+    {
+        /** @var \Cake\Database\Query\UpdateQuery $query */
+        $query = $this->newQuery(Query::TYPE_UPDATE);
+        $query = $query->update('articles')
+            ->set('name', 'mark')
+            ->where(['id' => 1])
+            ->comment('This is a test');
+        $result = $this->compiler->compile($query, $this->binder);
+        $this->assertSame('/* This is a test */ UPDATE articles SET name = :c0 WHERE id = :c1', $result);
+    }
+
+    public function testUpdateWithCommentInProd(): void
+    {
+        Configure::write('debug', false);
+        /** @var \Cake\Database\Query\UpdateQuery $query */
+        $query = $this->newQuery(Query::TYPE_UPDATE);
+        $query = $query->update('articles')
+            ->set('name', 'mark')
+            ->where(['id' => 1])
+            ->comment('This is a test');
+        $result = $this->compiler->compile($query, $this->binder);
+        $this->assertSame('UPDATE articles SET name = :c0 WHERE id = :c1', $result);
+    }
+
+    public function testDelete(): void
+    {
+        /** @var \Cake\Database\Query\DeleteQuery $query */
+        $query = $this->newQuery(Query::TYPE_DELETE);
+        $query = $query->delete()
+            ->from('articles')
+            ->where(['id !=' => 1]);
+        $result = $this->compiler->compile($query, $this->binder);
+        $this->assertSame('DELETE FROM articles WHERE id != :c0', $result);
+    }
+
+    public function testDeleteWithComment(): void
+    {
+        /** @var \Cake\Database\Query\DeleteQuery $query */
+        $query = $this->newQuery(Query::TYPE_DELETE);
+        $query = $query->delete()
+            ->from('articles')
+            ->where(['id !=' => 1])
+            ->comment('This is a test');
+        $result = $this->compiler->compile($query, $this->binder);
+        $this->assertSame('/* This is a test */ DELETE FROM articles WHERE id != :c0', $result);
+    }
+
+    public function testDeleteWithCommentInProd(): void
+    {
+        Configure::write('debug', false);
+        /** @var \Cake\Database\Query\DeleteQuery $query */
+        $query = $this->newQuery(Query::TYPE_DELETE);
+        $query = $query->delete()
+            ->from('articles')
+            ->where(['id !=' => 1])
+            ->comment('This is a test');
+        $result = $this->compiler->compile($query, $this->binder);
+        $this->assertSame('DELETE FROM articles WHERE id != :c0', $result);
+    }
+}
```

Comment:
This needs tests that actually execute the query against the database engine.

Reply:
Is it ok to just execute the query and assert the returning object type or should I add aditional checks?

Reply:
I was just looking for execution doesn't throw a db exception, nothing more.